README
NAME
    CPAN::Reporter - Adds CPAN Testers reporting to CPAN.pm

VERSION
    version 1.2011

SYNOPSIS
    From the CPAN shell:

      cpan> install Task::CPAN::Reporter
      cpan> reload cpan
      cpan> o conf init test_report

    Installing Task::CPAN::Reporter will pull in additional dependencies
    that new CPAN Testers will need.

    Advanced CPAN Testers with custom Test::Reporter::Transport setups may
    wish to install only CPAN::Reporter, which has fewer dependencies.

DESCRIPTION
    The CPAN Testers project captures and analyzes detailed results from
    building and testing CPAN distributions on multiple operating systems
    and multiple versions of Perl. This provides valuable feedback to module
    authors and potential users to identify bugs or platform compatibility
    issues and improves the overall quality and value of CPAN.

    One way individuals can contribute is to send a report for each module
    that they test or install. CPAN::Reporter is an add-on for the CPAN.pm
    module to send the results of building and testing modules to the CPAN
    Testers project. Full support for CPAN::Reporter is available in CPAN.pm
    as of version 1.92.

GETTING STARTED
  Installation
    The first step in using CPAN::Reporter is to install it using whatever
    version of CPAN.pm is already installed. CPAN.pm will be upgraded as a
    dependency if necessary.

      cpan> install CPAN::Reporter

    If CPAN.pm was upgraded, it needs to be reloaded.

      cpan> reload cpan

  Configuration
    If upgrading from a very old version of CPAN.pm, users may be prompted
    to renew their configuration settings, including the 'test_report'
    option to enable CPAN::Reporter.

    If not prompted automatically, users should manually initialize
    CPAN::Reporter support. After enabling CPAN::Reporter, CPAN.pm will
    automatically continue with interactive configuration of CPAN::Reporter
    options.

      cpan> o conf init test_report

    Users will need to enter an email address in one of the following
    formats:

      johndoe@example.com
      John Doe <johndoe@example.com>
      "John Q. Public" <johnqpublic@example.com>

    Users that are new to CPAN::Reporter should accept the recommended
    values for other configuration options.

    Users will be prompted to create a *Metabase profile* file that uniquely
    identifies their test reports. See "The Metabase" below for details.

    After completing interactive configuration, be sure to commit (save) the
    CPAN configuration changes.

      cpan> o conf commit

    See CPAN::Reporter::Config for advanced configuration settings.

   The Metabase
    CPAN::Reporter sends test reports to a server known as the Metabase.
    This requires an active Internet connection and a profile file. To
    create the profile, users will need to run "metabase-profile" from a
    terminal window and fill the information at the prompts. This will
    create a file called "metabase_id.json" in the current directory. That
    file should be moved to the ".cpanreporter" directory inside the user's
    home directory.

    Users with an existing metabase profile file (e.g. from another
    machine), should copy it into the ".cpanreporter" directory instead of
    creating a new one. Profile files may be located outside the
    ".cpanreporter" directory by following instructions in
    CPAN::Reporter::Config.

  Using CPAN::Reporter
    Once CPAN::Reporter is enabled and configured, test or install modules
    with CPAN.pm as usual.

    For example, to test the File::Marker module:

      cpan> test File::Marker

    If a distribution's tests fail, users will be prompted to edit the
    report to add additional information that might help the author
    understand the failure.

UNDERSTANDING TEST GRADES
    CPAN::Reporter will assign one of the following grades to the report:

    *   "pass" -- distribution built and tested correctly

    *   "fail" -- distribution failed to test correctly

    *   "unknown" -- distribution failed to build, had no test suite or
        outcome was inconclusive

    *   "na" --- distribution is not applicable to this platform and/or
        version of Perl

    In returning results of the test suite to CPAN.pm, "pass" and "unknown"
    are considered successful attempts to "make test" or "Build test" and
    will not prevent installation. "fail" and "na" are considered to be
    failures and CPAN.pm will not install unless forced.

    An error from Makefile.PL/Build.PL or make/Build will also be graded as
    "unknown" and a failure will be signaled to CPAN.pm.

    If prerequisites specified in "Makefile.PL" or "Build.PL" are not
    available, no report will be generated and a failure will be signaled to
    CPAN.pm.

PRIVACY WARNING
    CPAN::Reporter includes information in the test report about environment
    variables and special Perl variables that could be affecting test
    results in order to help module authors interpret the results of the
    tests. This includes information about paths, terminal, locale,
    user/group ID, installed toolchain modules (e.g. ExtUtils::MakeMaker)
    and so on.

    These have been intentionally limited to items that should not cause
    harmful personal information to be revealed -- it does *not* include
    your entire environment. Nevertheless, please do not use CPAN::Reporter
    if you are concerned about the disclosure of this information as part of
    your test report.

    Users wishing to review this information may choose to edit the report
    prior to sending it.

BUGS
    Please report any bugs or feature using the CPAN Request Tracker. Bugs
    can be submitted through the web interface at
    <http://rt.cpan.org/Dist/Display.html?Queue=CPAN-Reporter>

    When submitting a bug or request, please include a test-file or a patch
    to an existing test-file that illustrates the bug or desired feature.

SEE ALSO
    Information about CPAN::Testers:

    *   CPAN::Testers -- overview of CPAN Testers architecture stack

    *   <http://www.cpantesters.org> -- project home with all reports

    *   <http://wiki.cpantesters.org> -- documentation and wiki

    Additional Documentation:

    *   CPAN::Reporter::Config -- advanced configuration settings

    *   CPAN::Reporter::FAQ -- hints and tips

SUPPORT
  Bugs / Feature Requests
    Please report any bugs or feature requests through the issue tracker at
    <https://github.com/cpan-testers/CPAN-Reporter/issues>. You will be
    notified automatically of any progress on your issue.

  Source Code
    This is open source software. The code repository is available for
    public review and contribution under the terms of the license.

    <https://github.com/cpan-testers/CPAN-Reporter>

      git clone https://github.com/cpan-testers/CPAN-Reporter.git

AUTHOR
    David Golden <dagolden@cpan.org>

CONTRIBUTORS
    *   Alexandr Ciornii <alexchorny@gmail.com>

    *   Breno G. de Oliveira <garu@cpan.org>

    *   Christian Walde <walde.christian@googlemail.com>

    *   Kent Fredric <kentfredric@gmail.com>

    *   Matthew Musgrove <mr.muskrat@gmail.com>

COPYRIGHT AND LICENSE
    This software is Copyright (c) 2006 by David Golden.

    This is free software, licensed under:

      The Apache License, Version 2.0, January 2004



Changes
Revision history for Perl module CPAN::Reporter

1.2011    2014-03-16 22:27:56+01:00 Europe/Paris

  [ADDED]

  - Added support for future optional prerequisite reporting
    when CPAN.pm supports it

  [CHANGED]

  - Moved test distributions into corpus/ to avoid repetitive
    tarball unpacking

  - Modernized distribution meta and support files

1.2010    2013-04-12 15:35:38 Europe/London

  [NEW FEATURES]

  - new option 'retry_submission' tries to send the report one
    extra time in case it fails. Defaults to undef (don't retry)
    (Patch by Alexandr "chorny" Ciornii)

1.2009    2012-12-18 16:08:37 America/New_York

  [BUG FIXES]

  - Expanded prerequisite load check skip to all Acme::* modules; prereqs
    still need to be installed and of sufficient version, but we don't
    check that Acme modules load without error to avoid side effects.

1.2008    2012-12-15 07:26:11 America/New_York

  [BUG FIXES]

  - Don't try to check if Acme::Bleach can be loaded if it's listed
    as a prereq

1.2007    2012-12-10 15:13:22 America/New_York


  [BUG FIXES]

  - Recognize another form of "Perl is too low" message [CHORNY]

1.2006    2012-04-10 18:44:20 America/New_York

  [BUG FIXES]

  - Changed how CPAN::Reporter::PrereqCheck attempts to load modules to
    avoid module reloading, which caused some prereqs to show up as
    'broken' [rt.cpan.org #76394 and #75559]

  [OTHER]

  - Bumped IPC::Cmd prereq to 0.76 to avoid failures on some platforms

1.2005    2012-03-02 13:21:36 EST5EDT

  [BUG FIXES]

  - Canonicalise the config path returned by bsd_glob [rt.cpan.org #70504]
    [Kent Fredric]

1.2004    2012-02-13 16:25:07 EST5EDT

  [BUG FIXES]

  - Fix bug in testing tilde expansion on Windows
    [rt.cpan.org #72051]

1.2003    2011-12-06 11:07:04 EST5EDT

  [BUG FIXES]

  - Fix bug in transport class validation to allow multi-level
    class names [rt.cpan.org #73045]

1.2002    2011-08-10 17:52:08 America/New_York

  [BUG FIXES]

  - Fix failing tests on systems without
    Test::Reporter::Transport::Metabase
  - Minor test cleanups for use with 'prove'

1.2001    2011-08-07 05:05:58 America/New_York

  [BUG FIXES]

  - Fix failing test on Windows due to glob expansion of "~"
    [Christian Walde]

1.20      2011-08-05 13:17:20 America/New_York

  [BUG FIXES]

  - Removed Test::Reporter::Transport::Metabase as a prerequisite.
    New testers are recommended to install Task::CPAN::Reporter instead
    of CPAN::Reporter.

1.19_04   2011-06-29 17:39:25 America/New_York

  [NEW FEATURES]

  - Metabase profile generation will use as much of the provided
    email_from parameter as possible and will generate a random
    secret API key.  If user provides email_from in a standard form
    like C<< "John Q. Public" <jqp@example.com> >> then profile
    generation is entirely automatic without additional prompts
    from metabase-profile [David Golden]

  [DOCUMENTATION]

  - Swapped in Task::CPAN::Reporter in the SYNOPSIS [David Golden]

  [BUG FIXES]

  - Expands "~" into user directory before attempting to locate
    'relative' pathname in .cpanreporter directory [David Golden]

1.19_03   2011-06-28 14:44:19 America/New_York

  [NEW FEATURES]
  - During configuration, if 'Metabase' is the requested transport and
    a Metabase profile file is not found, configuration will offer to
    run metabase-profile to create it. [David Golden]

  - For Metabase transport (only), the 'id_file' may have a relative
    file path, treated as relative to the .cpanreporter directory.
    [David Golden]

  [BUG FIXES]
  - updated CPAN::Reporter to go with Metabase by default, instead of
    the deprecated SMTP transport system. This provides more up-to-date
    information for new testers on how to setup their systems and
    addresses RT#64316 (and maybe also RT#61735) [Breno G. de Oliveira]

  - Replaced use of Tee.pm with Capture::Tiny as the two don't play
    well together on some systems. [Christian Walde]

  [DOCUMENTATION]
  - Docs have been substantially re-written to reflect the new HTTP-based
    transport instead of email-based transport. [Breno G. de Oliveira and
    David Golden]

1.1902    2011-01-28 15:22:35 EST5EDT

  [BUG FIXES]

  - Updated Tee prereq to 0.14 to fix an instability on Windows

1.1901    2011-01-24 20:23:23 EST5EDT

  - Fixes some space and backslash escaping for Windows [Christian Walde]

1.1900    2011-01-21 10:55:28 EST5EDT

  - No changes from 1.18_06

1.18_06   2010-12-07 05:18:51 EST5EDT

  Fixed:

  - Reports with a "can't spawn" error from TAP::Parser::Iterator::Process
    are discarded instead of reporting FAIL

  Other:

  - CPAN::Reporter is now released using Dist::Zilla.  This required
    some changes to various files help dzil find/skip various bits of data.
    It should have no runtime effect.

1.18_05 Sun Oct 24 21:25:22 EDT 2010

  Fixed:

  - Fixed error in calling prerequisite detection script when there
    are spaces in the path [RURBAN]

1.18_04 NOT RELEASED TO CPAN

  Changed:

  - Executed processes are given their own process group and timeouts are
    now handled by sending SIGKILL to the entire child process group.
    This is an attempt to address RT#61912

1.1803 Thu Sep 30 21:49:10 EDT 2010

  Fixed:

  - Fixed t/13_record_command.t to reflect removal of Proc::ProcessTable
    as a conditional dependency and fixed a resulting bug in the test
  
  - Removed mention of Proc::ProcessTable in CPAN::Reporter::Config

1.1802 Wed Sep 29 23:19:15 EDT 2010

  Changed:

  - Given the migration to CT2.0, the maximum report length before
    truncation has been increased from 50 to 1,000 kB.

  - Removed dependency on Proc::ProcessTable for using a command timeout
    on non-Win32 platforms.

1.1801 Fri Sep 10 16:34:29 EDT 2010

  Changed:

  - Default config directory now uses File::HomeDir->my_home on
    windows instead of ->my_documents.  If that directory doesn't exist,
    it will fall back to ->my_documents.


t/Frontend.pm
#
# This file is part of CPAN-Reporter
#
# This software is Copyright (c) 2006 by David Golden.
#
# This is free software, licensed under:
#
#   The Apache License, Version 2.0, January 2004
#
package t::Frontend;
use strict;
BEGIN{ if (not $] < 5.006) { require warnings; warnings->import } }

use ExtUtils::MakeMaker ();

BEGIN {
    $INC{"CPAN.pm"} = 1; #fake load
    $CPAN::VERSION = 999;
}

package CPAN::Shell;

sub myprint {
    shift;
    print @_;
}

sub mywarn {
    shift;
    print @_;
}

sub colorable_makemaker_prompt {
    goto \&ExtUtils::MakeMaker::prompt;
}

package CPAN;

$CPAN::Frontend = $CPAN::Frontend = "CPAN::Shell";

1;


t/MockHomeDir.pm
#
# This file is part of CPAN-Reporter
#
# This software is Copyright (c) 2006 by David Golden.
#
# This is free software, licensed under:
#
#   The Apache License, Version 2.0, January 2004
#
package t::MockHomeDir;
use strict;
BEGIN{ if (not $] < 5.006) { require warnings; warnings->import } }
use File::Spec;
use File::Temp qw/tempdir/;

$INC{"File/HomeDir.pm"} = 1; # fake load

# tester might have these set -- we need them unset so we use our
# own temporary dir & config during our tests
delete $ENV{PERL_CPAN_REPORTER_DIR};
delete $ENV{PERL_CPAN_REPORTER_CONFIG};

my $temp_home = tempdir( 
    "CPAN-Reporter-testhome-XXXXXXXX", TMPDIR => 1, CLEANUP => 1 
) or die "Couldn't create a temporary config directory: $!\nIs your temp drive full?";

my $home_dir = File::Spec->rel2abs( $temp_home );

sub home_dir { $home_dir }

package File::HomeDir;
our $VERSION = 999;
sub my_documents { return $home_dir };
sub my_home { return $home_dir };
sub my_data { return $home_dir };

1;



Build.PL
use strict;
use Module::Build;
use File::Spec;

my $class = Module::Build->subclass(
    class => "Module::Build::WikiDoc",
    code => <<'SUBCLASS', );

    sub ACTION_wikidoc {
        my $self = shift;
        eval "use Pod::WikiDoc";
        if ( $@ eq '' ) {
            my $parser = Pod::WikiDoc->new({ 
                comment_blocks => 1,
                keywords => { VERSION => $self->dist_version },
            });
            for my $src ( keys %{ $self->find_pm_files() } ) {
                (my $tgt = $src) =~ s{\.pm$}{.pod};
                $parser->filter( {
                    input   => $src,
                    output  => $tgt,
                });
                print "Creating $tgt\n";
                $tgt =~ s{\\}{/}g;
                $self->_add_to_manifest( 'MANIFEST', $tgt );
            }
        }
        else {
            warn "Pod::WikiDoc not available. Skipping wikidoc.\n";
        }
    }

    sub ACTION_test {
        my $self = shift;
        my $missing_pod;
        for my $src ( keys %{ $self->find_pm_files() } ) {
            (my $tgt = $src) =~ s{\.pm$}{.pod};
            $missing_pod = 1 if ! -e $tgt;
        }
        if ( $missing_pod ) {
            $self->depends_on('wikidoc');
            $self->depends_on('build');
        }
        $self->SUPER::ACTION_test;
    }

    sub ACTION_testpod {
        my $self = shift;
        $self->depends_on('wikidoc');
        $self->SUPER::ACTION_testpod;
    }

    sub ACTION_distdir {
        my $self = shift;
        $self->depends_on('wikidoc');
        $self->SUPER::ACTION_distdir;
    }

SUBCLASS

$class->new( 
    module_name         => 'CPAN::Reporter',
    dist_author         => 'David A. Golden <dagolden@cpan.org>',
    license             => 'perl',
    create_readme       => 1,
    create_makefile_pl  => 'traditional',
    requires        => {
        # module requirements here
        'Config::Tiny'          => 2.08,
        'CPAN'                  => 1.88,
        'ExtUtils::MakeMaker'   => 6.26, # sane ExtUtils::Command::MM
        'File::Basename'        => 0,
        'File::Copy::Recursive' => 0.28,
        'File::HomeDir'         => 0.58, 
        'File::Path'            => 0,
        'File::pushd'           => 0.32,
        'File::Spec'            => 3.19,
        'File::Temp'            => 0.16,
        'IO::CaptureOutput'     => 1.03,  # build_requires
        'Probe::Perl'           => 0,
        'Tee'                   => 0.13,
        'Test::More'            => 0.62,   
        'Test::Reporter'        => 1.27,
    },
    meta_add            => { 
        no_index => { 
            file => [ qw{ 
                lib/CPAN/Reporter/FAQ.pm
            } ],
            directory => [ 'examples' ],
        } 
    },
)->create_build_script;


lib/CPAN/Reporter/API.pm
use strict; # make CPANTS happy
package CPAN::Reporter::API;
our $VERSION = '1.2011'; # VERSION

1;

# ABSTRACT: Programmer's interface to CPAN::Reporter

__END__

=pod

=encoding UTF-8

=head1 NAME

CPAN::Reporter::API - Programmer's interface to CPAN::Reporter

=head1 VERSION

version 1.2011

=head1 FUNCTIONS

CPAN::Reporter provides only a few public function for use within CPAN.pm.
They are not imported during C<<< use >>>.  Ordinary users will never need them.

=head2 C<<< configure() >>>

  CPAN::Reporter::configure();

Prompts the user to edit configuration settings stored in the CPAN::Reporter
C<<< config.ini >>> file.  It will create the configuration file if it does not exist.
It is automatically called by CPAN.pm when initializing the 'test_report'
option, e.g.:

  cpan> o conf init test_report

=head2 C<<< record_command() >>>

  ($output, $exit_value) = CPAN::Reporter::record_command( $cmd, $secs );

Takes a command to be executed via system(), but wraps and tees it to
show the output to the console, capture the output, and capture the
exit code.  Returns an array reference of output lines (merged STDOUT and
STDERR) and the return value from system().  Note that this is C<<< $? >>>, so the
actual exit value of the command will need to be extracted as described in
L<perlvar>.

If the command matches E<sol>\b(?:MakefileE<verbar>Build)\.PL\bE<sol>, then L<Devel::Autoflush>
is added to C<<< PERL5OPT >>> to force autoflushing of user prompts.

If the command includes a pipe character ('E<verbar>'), only the part of the 
command prior to the pipe will be wrapped and teed.  The pipe will be
applied to the execution of the wrapper script.  This is essential to 
capture the exit value of the command and should be otherwise transparent.

If a non-zero C<<< $secs >>> argument is provided, the command will be run with a
timeout of C<<< $secs >>> (seconds) if the appropriate process management modules are
available.  On Win32, L<Win32::Job> must be available; on Unix, L<Proc::Killfam>
must be available.  Otherwise, code will fall-back to running without a
timeout.

Also, on Win32, the first space-separated element of the command must be
absolute, or else ".exe" will be appended and the PATH searched for a matching
command.

If the attempt to record fails, a warning will be issued and one or more of 
C<<< $output >>> or C<<< $exit_value >>> will be undefined.

=head2 C<<< grade_make() >>>

  CPAN::Reporter::grade_make( $dist, $command, \@output, $exit);

Given a CPAN::Distribution object, the system command used to build the
distribution (e.g. "make", "perl Build"), an array of lines of output from the
command and the exit value from the command, C<<< grade_make() >>> determines a grade
for this stage of distribution installation.  If the grade is "pass",
C<<< grade_make() >>> does B<not> send a CPAN Testers report for this stage and returns
true to signal that the build was successful.  Otherwise, a CPAN Testers report
is sent and C<<< grade_make() >>> returns false.

=head2 C<<< grade_PL() >>>

  CPAN::Reporter::grade_PL( $dist, $command, \@output, $exit);

Given a CPAN::Distribution object, the system command used to run Makefile.PL
or Build.PL (e.g. "perl Makefile.PL"), an array of lines of output from the
command and the exit value from the command, C<<< grade_PL() >>> determines a grade
for this stage of distribution installation.  If the grade is "pass",
C<<< grade_PL() >>> does B<not> send a CPAN Testers report for this stage and returns
true to signal that the Makefile.PL or Build.PL ran successfully.  Otherwise, a
CPAN Testers report is sent and C<<< grade_PL() >>> returns false.

=head2 C<<< grade_test() >>>

  CPAN::Reporter::grade_test( $dist, $command, \@output, $exit);

Given a CPAN::Distribution object, the system command used to run tests (e.g.
"make test"), an array of lines of output from testing and the exit value from
the system command, C<<< grade_test() >>> determines a grade for distribution tests.
A CPAN Testers report is then sent unless specified prerequisites were not
satisfied, in which case the report is discarded.  This function returns true
if the grade is "pass" or "unknown" and returns false, otherwise.

=head2 C<<< test() >>> -- DEPRECATED

  CPAN::Reporter::test( $cpan_dist, $system_command );

This function is maintained for backwards compatibility.  It effectively 
wraps the functionality of C<<< record_command() >>> and C<<< grade_test() >>> into
a single function call. It takes a CPAN::Distribution object and the system
command to run distribution tests.

=head1 SEE ALSO

=over

=item *

L<CPAN::Reporter>

=item *

L<CPAN::Reporter::Config>

=item *

L<CPAN::Reporter::FAQ>

=back

=head1 AUTHOR

David Golden <dagolden@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is Copyright (c) 2006 by David Golden.

This is free software, licensed under:

  The Apache License, Version 2.0, January 2004

=cut


lib/CPAN/Reporter/FAQ.pm
use strict; # make CPANTS happy
package CPAN::Reporter::FAQ;
our $VERSION = '1.2011'; # VERSION

1;

# ABSTRACT: Answers and tips for using CPAN::Reporter

__END__

=pod

=encoding UTF-8

=head1 NAME

CPAN::Reporter::FAQ - Answers and tips for using CPAN::Reporter

=head1 VERSION

version 1.2011

=head1 REPORT GRADES

=head2 Why did I receive a report? 

Historically, CPAN Testers was designed to have each tester send a copy of
reports to authors.  This philosophy changed in September 2008 and CPAN Testers
tools were updated to no longer copy authors, but some testers may still be
using an older version.

=head2 Why was a report sent if a prerequisite is missing?

As of CPAN::Reporter 0.46, FAIL and UNKNOWN reports with unsatisfied 
prerequisites are discarded.  Earlier versions may have sent these reports 
out by mistake as either an NA or UNKNOWN report.

PASS reports are not discarded because it may be useful to know when tests
passed despite a missing prerequisite.  NA reports are sent because information
about the lack of support for a platform is relevant regardless of
prerequisites.

=head1 SENDING REPORTS

=head2 Why did I get an error sending a test report?

Historically, test reports were sent via ordinary email.
The most common reason for errors sending a report back then was that
many Internet Service Providers (ISP's) would block
outbound SMTP (email) connections as part of their efforts to fight spam.

Since 2010, test reports are sent to the CPAN Testers Metabase over HTTPS. The
most common reason for failures are systems which upgraded CPAN::Reporter but
are still configured to use the deprecated and unsupported email system instead
of Metabase for transport.

If you are unsure which transport mechanism you're using, look for the
C<<< transport >>> rule in the C<<< .cpanreporter/config.ini >>> file, in the
user's home directory.  See L<CPAN::Reporter::Config> for details on how
to set the C<<< transport >>> option for Metabase.

Other errors could be caused by the absence of the
C<<< .cpanreporter/metabase_id.json >>> file in the user's home directory. This file
should be manually created prior to sending any reports, via the
C<<< metabase-profile >>> program. Simply run it and fill in the information
accordingly, and it will create the C<<< metabase_id.json >>> file for you. Move that
file to your C<<< .cpanreporter >>> directory and you're all set.

If you experience intermittent network issues, you can set the
'retry_submission' option to make a second attempt at sending the report
a few seconds later, in case the first one fails. This could be useful for
extremely slow connections.

Finally, lack of Internet connection or firewall filtering will prevent
the report from reaching the CPAN Testers servers. If you are experiencing
HTTPS issues or messages complaining about SSL modules, try installing
the L<LWP::Protocol::https> module and trying again. If all fails, you
may still change the transport uri to use HTTP instead of HTTPS, though
this is I<not> recommended.

=head2 Why didn't my test report show up on CPAN Testers?

There is a delay between the time reports are sent to the Metabase and when
they they appear on the CPAN Testers website. There is a further delay before
summary statistics appear on search.cpan.org.  If your reports do not appear
after 24 hours, please contact the cpan-testers-discuss email list
(L<http://lists.perl.org/list/cpan-testers-discuss.html>) or join the
C<<< #cpantesters-discuss >>> IRC channel on C<<< irc.perl.org >>>.

=head1 CPAN TESTERS

=head2 Where can I find out more about CPAN Testers?

A good place to start is the CPAN Testers Wiki: 
L<http://wiki.cpantesters.org/>

=head2 Where can I find statistics about reports sent to CPAN Testers?

CPAN Testers statistics are compiled at L<http://stats.cpantesters.org/>

=head2 How do I make sure I get credit for my test reports?

To get credit in the statistics, use the same Metabase profile file
and the same email address wherever you run tests.

=head1 SEE ALSO

=over

=item *

L<CPAN::Testers>

=item *

L<CPAN::Reporter>

=item *

L<Test::Reporter>

=back

=head1 AUTHOR

David Golden <dagolden@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is Copyright (c) 2006 by David Golden.

This is free software, licensed under:

  The Apache License, Version 2.0, January 2004

=cut


Makefile.PL

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.014.
use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker 6.17;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Adds CPAN Testers reporting to CPAN.pm",
  "AUTHOR" => "David Golden <dagolden\@cpan.org>",
  "BUILD_REQUIRES" => {},
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.17"
  },
  "DISTNAME" => "CPAN-Reporter",
  "EXE_FILES" => [],
  "LICENSE" => "apache",
  "NAME" => "CPAN::Reporter",
  "PREREQ_PM" => {
    "CPAN" => "1.94",
    "CPAN::Version" => 0,
    "Capture::Tiny" => 0,
    "Carp" => 0,
    "Config::Tiny" => "2.08",
    "Data::Dumper" => 0,
    "Devel::Autoflush" => "0.04",
    "Exporter" => 0,
    "ExtUtils::MakeMaker" => "6.36",
    "Fcntl" => 0,
    "File::Basename" => 0,
    "File::Find" => 0,
    "File::Glob" => 0,
    "File::HomeDir" => "0.58",
    "File::Path" => 0,
    "File::Spec" => "3.19",
    "File::Temp" => "0.16",
    "IO::File" => 0,
    "IPC::Cmd" => "0.76",
    "Parse::CPAN::Meta" => 0,
    "Probe::Perl" => 0,
    "Test::Reporter" => "1.54",
    "constant" => 0,
    "strict" => 0,
    "vars" => 0
  },
  "TEST_REQUIRES" => {
    "Archive::Tar" => "1.54",
    "File::Copy::Recursive" => "0.35",
    "File::Spec::Functions" => 0,
    "File::pushd" => "0.32",
    "IO::CaptureOutput" => "1.03",
    "List::Util" => 0,
    "Test::Harness" => 0,
    "Test::More" => "0.62",
    "version" => 0,
    "warnings" => 0
  },
  "VERSION" => "1.2011",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Archive::Tar" => "1.54",
  "CPAN" => "1.94",
  "CPAN::Version" => 0,
  "Capture::Tiny" => 0,
  "Carp" => 0,
  "Config::Tiny" => "2.08",
  "Data::Dumper" => 0,
  "Devel::Autoflush" => "0.04",
  "Exporter" => 0,
  "ExtUtils::MakeMaker" => "6.36",
  "Fcntl" => 0,
  "File::Basename" => 0,
  "File::Copy::Recursive" => "0.35",
  "File::Find" => 0,
  "File::Glob" => 0,
  "File::HomeDir" => "0.58",
  "File::Path" => 0,
  "File::Spec" => "3.19",
  "File::Spec::Functions" => 0,
  "File::Temp" => "0.16",
  "File::pushd" => "0.32",
  "IO::CaptureOutput" => "1.03",
  "IO::File" => 0,
  "IPC::Cmd" => "0.76",
  "List::Util" => 0,
  "Parse::CPAN::Meta" => 0,
  "Probe::Perl" => 0,
  "Test::Harness" => 0,
  "Test::More" => "0.62",
  "Test::Reporter" => "1.54",
  "constant" => 0,
  "strict" => 0,
  "vars" => 0,
  "version" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



